controllers/chat/initChatFromPost.js

```diff
@@ -186,16 +186,26 @@ const initChatFromPost = async (req, res) => {
         members
       });
     } else {
-      let anonUserInfoFromPost = {};
+      let anonUserInfoFromSouce = {};
       channel_type = CHANNEL_TYPE.ANONYMOUS;
       if (targetUserModel.is_anonymous) {
-        let anon_post_detail = await Getstream.feed.getPlainFeedById(postId);
-        anonUserInfoFromPost = {
-          anon_user_info_color_code: anon_post_detail.anon_user_info_color_code,
-          anon_user_info_color_name: anon_post_detail.anon_user_info_color_name,
-          anon_user_info_emoji_code: anon_post_detail.anon_user_info_emoji_code,
-          anon_user_info_emoji_name: anon_post_detail.anon_user_info_emoji_name
-        };
+        if (source === 'post') {
+          let anon_post_detail = await Getstream.feed.getPlainFeedById(postId);
+          anonUserInfoFromSouce = {
+            anon_user_info_color_code: anon_post_detail.anon_user_info_color_code,
+            anon_user_info_color_name: anon_post_detail.anon_user_info_color_name,
+            anon_user_info_emoji_code: anon_post_detail.anon_user_info_emoji_code,
+            anon_user_info_emoji_name: anon_post_detail.anon_user_info_emoji_name
+          };
+        } else {
+          let anon_comment_detail = await Getstream.feed.getReactionById(commentId);
+          anonUserInfoFromSouce = {
+            anon_user_info_color_code: anon_comment_detail.data.anon_user_info_color_code,
+            anon_user_info_color_name: anon_comment_detail.data.anon_user_info_color_name,
+            anon_user_info_emoji_code: anon_comment_detail.data.anon_user_info_emoji_code,
+            anon_user_info_emoji_name: anon_comment_detail.data.anon_user_info_emoji_name
+          };
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for fetching anonymous user information from either a post or a comment is correct. However, there's a potential issue with error handling. If the `Getstream.feed.getPlainFeedById(postId)` or `Getstream.feed.getReactionById(commentId)` calls fail, it could lead to an unhandled promise rejection. It would be better to wrap these calls in a try-catch block to handle any potential errors.
```diff
-          if (source === 'post') {
-            let anon_post_detail = await Getstream.feed.getPlainFeedById(postId);
-            anonUserInfoFromSouce = {
-              anon_user_info_color_code: anon_post_detail.anon_user_info_color_code,
-              anon_user_info_color_name: anon_post_detail.anon_user_info_color_name,
-              anon_user_info_emoji_code: anon_post_detail.anon_user_info_emoji_code,
-              anon_user_info_emoji_name: anon_post_detail.anon_user_info_emoji_name
-            };
-          } else {
-            let anon_comment_detail = await Getstream.feed.getReactionById(commentId);
-            anonUserInfoFromSouce = {
-              anon_user_info_color_code: anon_comment_detail.data.anon_user_info_color_code,
-              anon_user_info_color_name: anon_comment_detail.data.anon_user_info_color_name,
-              anon_user_info_emoji_code: anon_comment_detail.data.anon_user_info_emoji_code,
-              anon_user_info_emoji_name: anon_comment_detail.data.anon_user_info_emoji_name
-            };
-          }
+          try {
+            if (source === 'post') {
+              let anon_post_detail = await Getstream.feed.getPlainFeedById(postId);
+              anonUserInfoFromSouce = {
+                anon_user_info_color_code: anon_post_detail.anon_user_info_color_code,
+                anon_user_info_color_name: anon_post_detail.anon_user_info_color_name,
+                anon_user_info_emoji_code: anon_post_detail.anon_user_info_emoji_code,
+                anon_user_info_emoji_name: anon_post_detail.anon_user_info_emoji_name
+              };
+            } else {
+              let anon_comment_detail = await Getstream.feed.getReactionById(commentId);
+              anonUserInfoFromSouce = {
+                anon_user_info_color_code: anon_comment_detail.data.anon_user_info_color_code,
+                anon_user_info_color_name: anon_comment_detail.data.anon_user_info_color_name,
+                anon_user_info_emoji_code: anon_comment_detail.data.anon_user_info_emoji_code,
+                anon_user_info_emoji_name: anon_comment_detail.data.anon_user_info_emoji_name
+              };
+            }
+          } catch (error) {
+            console.error(`Failed to fetch anonymous user info from ${source}:`, error);
+            throw error;
+          }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->